examples/chip-tool/commands/clusters/Commands.h

```diff
@@ -11919,6 +11939,212 @@ class ReadNetworkCommissioningClusterRevision : public ModelCommand
+                                           mCurrentVersion);
+    }
+
+private:
+    chip::Callback::Callback<DefaultSuccessCallback> * onSuccessCallback =
+        new chip::Callback::Callback<DefaultSuccessCallback>(OnDefaultSuccessResponse, this);
+    chip::Callback::Callback<DefaultFailureCallback> * onFailureCallback =
+        new chip::Callback::Callback<DefaultFailureCallback>(OnDefaultFailureResponse, this);
+    char * mUpdateToken;
+    uint32_t mCurrentVersion;
+};
+
+/*
+ * Command QueryImage
+ */
+class OtaSoftwareUpdateServerQueryImage : public ModelCommand
+{
+public:
+    OtaSoftwareUpdateServerQueryImage() : ModelCommand("query-image")
+    {
+        AddArgument("vendorId", 0, UINT16_MAX, &mVendorId);
+        AddArgument("productId", 0, UINT16_MAX, &mProductId);
+        AddArgument("imageType", 0, UINT16_MAX, &mImageType);
+        AddArgument("hardwareVersion", 0, UINT16_MAX, &mHardwareVersion);
+        AddArgument("currentVersion", 0, UINT32_MAX, &mCurrentVersion);
+        AddArgument("protocolsSupported", 0, UINT8_MAX, &mProtocolsSupported);
+        AddArgument("location", &mLocation);
+        AddArgument("clientCanConsent", 0, UINT8_MAX, &mClientCanConsent);
+        AddArgument("metadataForServer", &mMetadataForServer);
+        ModelCommand::AddArguments();
+    }
+    ~OtaSoftwareUpdateServerQueryImage()
+    {
+        delete onSuccessCallback;
+        delete onFailureCallback;
+    }
+
+    CHIP_ERROR SendCommand(ChipDevice * device, uint8_t endpointId) override
+    {
+        ChipLogProgress(chipTool, "Sending cluster (0x0029) command (0x00) on endpoint %" PRIu16, endpointId);
+
+        chip::Controller::OtaSoftwareUpdateServerCluster cluster;
+        cluster.Associate(device, endpointId);
+        return cluster.QueryImage(onSuccessCallback->Cancel(), onFailureCallback->Cancel(), mVendorId, mProductId, mImageType,
+                                  mHardwareVersion, mCurrentVersion, mProtocolsSupported,
+                                  chip::ByteSpan(chip::Uint8::from_char(mLocation), strlen(mLocation)), mClientCanConsent,
+                                  chip::ByteSpan(chip::Uint8::from_char(mMetadataForServer), strlen(mMetadataForServer)));
+    }
+
+private:
+    chip::Callback::Callback<OtaSoftwareUpdateServerClusterQueryImageResponseCallback> * onSuccessCallback =
+        new chip::Callback::Callback<OtaSoftwareUpdateServerClusterQueryImageResponseCallback>(
+            OnOtaSoftwareUpdateServerClusterQueryImageResponse, this);
+    chip::Callback::Callback<DefaultFailureCallback> * onFailureCallback =
+        new chip::Callback::Callback<DefaultFailureCallback>(OnDefaultFailureResponse, this);
+    uint16_t mVendorId;
+    uint16_t mProductId;
+    uint16_t mImageType;
+    uint16_t mHardwareVersion;
+    uint32_t mCurrentVersion;
+    uint8_t mProtocolsSupported;
+    char * mLocation;
+    uint8_t mClientCanConsent;
+    char * mMetadataForServer;
```

Comment:
This is not a char*, it is an octet string, which may likely contain internal nulls

Reply:
@mrjerryjohns says this is intended (for now?).

Reply:
Not sure about internal nulls though